src/lib/storage/defragment_reference_table.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+
+#include "types.hpp"
+
+namespace opossum {
+
+class Table;
+
+/**
+ * Creates a new Table, merging consecutive small Chunks into one bigger Chunk if possible, to decrease the overall
+ * number of Chunks. The order of rows is preserved.
```

Comment:
I feel like wewill have to revisit this once we look at NUMA optimizations. This would destroy any locality by the MPSM Join. But until we have an idea there, this in fine.